### PR TITLE
Remove deprecated prop types, add TS typings, add new props to customize display behavior 

### PR DESCRIPTION
**Pull request by hochem**
Added new props:
- errorMessageHidden (bool): Show/Hide the error message when in error state (false by default)
- inactivePlaceholder (bool): Show/Hide the placeholder when unfocused/inactive 
- Added TS typings

### CODE REVIEW

File: src/components/affix/index.js [hunk 1]
```diff
@@ -59,9 +54,7 @@ export default class Affix extends PureComponent {
     }
 
     return (
-      <Animated.View style={[styles.container, containerStyle]}>
-        <Animated.Text style={[style, textStyle]}>{children}</Animated.Text>
-      </Animated.View>
+      <Text style={[styles.container, containerStyle, style, textStyle]}>{children}</Text>
```
Comment (serratus) [inline src/components/affix/index.js:54]: This change also means that we have to have the `label` always positioned above the input field, even in cases where it's not focused? Correct?

**Review by serratus — APPROVED** [review]

File: src/components/field/index.js [hunk 2]
```diff
@@ -452,14 +473,11 @@ export default class TextField extends PureComponent {
   inputProps() {
     let store = {};
 
-    for (let key in TextInput.propTypes) {
+    for (let key in this.props) {
```
Comment (serratus) [inline src/components/field/index.js:460]: Does that not create a lot of "wrong" unexpected prop-types issues?

File: src/components/affix/index.js (hunk 1 above)
Comment (hochem) [reply src/components/affix/index.js:54]: No, the affix animation is now done on a parent container which also animates the text input. It shouldn't cause issues with the label, only downside is that the textinput and the affix can't be animated separately (but we currently don't have that usecase).

File: src/components/field/index.js (hunk 2 above)
Comment (hochem) [reply src/components/field/index.js:460]: Adapted the check to not pass any of the react-native-material-textfield props to RN TextInput

**Pull request merged**